modules/common/secret/secret.go

```diff
@@ -385,3 +386,70 @@ func GetDataFromSecret(
 
 	return data, ctrl.Result{}, nil
 }
+
+// EnsureSecretData - ensures that the Secret object exists and the expected fields
+// are in the Secret. It returns a hash of the values of the expected fields.
+func EnsureSecretData(
+	ctx context.Context,
+	secretName types.NamespacedName,
+	expectedFields []string,
+	reader client.Reader,
+	Updater condition.Updater,
+	requeueTimeout time.Duration,
+) (string, ctrl.Result, error) {
+	secret := &corev1.Secret{}
+	err := reader.Get(ctx, secretName, secret)
+	if err != nil {
+		if k8s_errors.IsNotFound(err) {
+			// TODO(gibi): Change the message to state which input
+			// (i.e. Secret with a given name) is missing
+			Updater.Set(condition.FalseCondition(
```

Comment:
I am not sure if we want to update the condition in the lib-common funcs. It might better/easier to just return the error message and to the condition update in the caller, but want to hear other peoples opinions.

We did something like that in the OSP Director Operator where we set condition on a more fine level, but need to be careful to not have to detailed and maybe changing in information in the message. This can then result in unnecessary object updates as the status changes without the need.

Reply:
Good point. I think this would be a first time moving condition updating to lib-common. So we need to think about it.

Reply:
should I remove the conditions?

Reply:
I'm torn. Half of the logic we generalizing here it the actual condition setting logic. So by removing that we force the callers to duplicate condition handling logic in each operator. For me the question is do we know that this condition setting logic is generic for all operators that depends on Secrets? If so then I think it is fine to put the logic into lib-common.

Reply:
i would remove the condtion updating for now and just pass back the error as martin suggested.

we can always creat a generic function that will take a condition and a function to call and set the condition if an error is return or something like that later. and its not hard to refactor nova to just do the condition setting on our side so we can defer that to the caller.

Reply:
I'd remove setting the condition and set them, like we do it right now in the controller of the operator as you might not want to get a condition set in all the cases you'd use this func.

Reply:
OK, I'm fine with removing the condition settings from here.